### ISSUE TITLE
APIGatewayProxyRequestEvent does not have authorizer

### ISSUE THREAD

**Issue by oharaandrew314** [aws/aws-lambda-java-libs#33]
The `APIGatewayProxyRequestEvent` `requestContext` object does not have the `authorizer` object.  With the new addition of the authorizer context, I am unsure how to structure a new `Authorizer` object.

It is entirely possible to have an authorizer json blob like:
```
{
    "principalId": "123", // Required by authorizer
    "roles": "role1,role2",  // Additional arbitrary parameter from authorizer context
    "likes_pie": true // Additional arbitrary parameter from authorizer context
}
```

The `principalId` is always part of the `authorizer` blob, and thus should be a String field on a new `Authorizer` object.  However, the additional arbitrary parameters provided by the authorizer context should be a map.  This won't (de)serialize 1:1 between JSON and Java however.

**Comment by onema** [comment]
I second this, any news on this?

**Comment by oharaandrew314** [comment]
Yes, I made a PR in #39

**Comment by oharaandrew314** [comment]
@onema, in the meantime, you can do what I do and redefine the file in your project.  If you use the exact same package, it should override the version in the library.

**Comment by avanathan** [comment]
@oharaandrew314 Could you pls. reopen PR #39 ? Everyone making changes in their code repo for a generic change is not a scalable solution. Thanks!

**Comment by oharaandrew314** [comment]
@avanathan I'm no longer using lambdas with Java, so I believe I'd be a poor champion for the PR.      If you wish, feel free to copy my changes and make your own PR.

 You may have the modify the map to use Object though.  It does seem that you can get non-string values in some cases.

**Comment by avanathan** [comment]
@oharaandrew314 Thanks. I did open a PR - https://github.com/aws/aws-lambda-java-libs/pull/77, but kept it `Map<String, String>`. Mainly because there are other collections using `String` key-value. Also, as per my understanding they other non-String objects will get converted as Strings before they are added to this map. Hope this rationale makes sense.

**Comment by carlzogh** [comment]
This was released as part of version `2.2.6` - thanks for your PR @avanathan!
